### DISCUSSION TITLE
Problems with unsupervised classification GPU load

### DISCUSSION THREAD

**Discussion by patrickmonari** [DrCoffey/DeepSqueak#198] [Q&A]
Hi! I'm trying to run the autoencoder + call features unsupervised classifier, but it keeps aborting because of insufficient memory on my GPU (RTX 3060). Is there a way to run this classifier without buying a better graphics card, or will I end up just having to use the call features only, instead of the autoencoder?

Thanks!
Patrick

**Comment by DrCoffey** [comment]
Hey Patrick, can you give me a few more details? Does it crash during VAE training? How big is the training set? Does it work with half the data?

I am using a 2060 for most things and haven't run into any memory issues yet.

**Reply by patrickmonari** [reply]
Hey! 

It seems to be crashing during training - it can load the files but won't proceed after that. It works on a smaller set (the one I'm training on is probably ~5,000 calls). I have no idea what I'm talking about, but is there a way to implement parallel processing for this kind of thing?

![image](https://github.com/DrCoffey/DeepSqueak/assets/61850735/d4efeb5f-7790-4875-a5cd-6316e585dfb9)

![image](https://github.com/DrCoffey/DeepSqueak/assets/61850735/4213702f-3030-4fdc-9efa-365fc0a32d8c)

Cheers,
Patrick

**Reply by patrickmonari** [reply]
Figured out the issue! Training automatically includes rejected calls, and there were *a lot* in this set. Pre-removing them reduced the size of the training set by an order of magnitude lol.

It might be nice to have a way to batch-remove rejected calls from detection files. Unless I'm missing something, it seems like the only way to do it right now is by opening each file one-by-one.

Cheers,
Patrick

**Comment by DrCoffey** [comment]
Hey, there is a tool in the tools menu for removing rejected calls. Clustering should do that automatically though; I may have dropped that when integrating the two versions. Rejected calls really complicate clustering.
________________________________
